File: Python/fib-iterative.py
import time
import itertools

# Iterative Fibonacci Sequence
def iterativeFib(n):
    if n < 2:
        return n

    a = 0
    b = 1
    i = 0
    
    while i < n:
        temp = a
        a = b
        b = temp + b
        i += 1
        
    return a

start_time = time.time()
for _, i in itertools.product(range (10000000), range(25)):
    iterativeFib(i)

print(f"--- {time.time() - start_time} seconds ---")

File: Rust/fib-iterative.rs
use std::time::{Instant};

fn iterative_fib(n: i32) -> i32 {
    if n < 2 {
        return n
    }

    let mut a = 0;
    let mut b = 1;
    let mut i = 0;
    
    while i < n {
        let temp = a;
        a = b;
        b = temp + b;
        i = i + 1;
    }
    return a;
}

fn main() {
    let start = Instant::now();

    for _ in 0..10000000{
        for i in 0..25 {
            iterative_fib(i);
        }
    }

    let duration = start.elapsed();

    println!("--- {:?} ---", duration);
}

